### ISSUE TITLE
Make MUI-datatables WCAG 2.1 compatible

### ISSUE THREAD

**Issue by krisgoks** [gregnb/mui-datatables#1361]
##Current behaviour

Thank you for this wonderful library.  I was just testing this library using WAVE evaluation tool on Google Chrome https://wave.webaim.org/. 
What WAVE does is: (taken from their website) "WAVE can identify many accessibility and Web Content Accessibility Guideline (WCAG) errors, but also facilitates human evaluation of web content."
When I do the test the result reports an accessibility error for the default MUI-datatable selectable checkbox form control, as it has a missing form label

## Expected Behavior

I do not know how to fix it myself or give you very meaningful info (I'm new to React) but the below is what the report tells me:

###What It Means
A form control does not have a corresponding label.
###Why It Matters
If a form control does not have a properly associated text label, the function or purpose of that form control may not be presented to screen reader users. Form labels also provide visible descriptions and larger clickable targets for form controls.
###How to Fix It
If a text label for a form control is visible, use the <label> element to associate it with its respective form control. If there is no visible label, either provide an associated label, add a descriptive title attribute to the form control, or reference the label(s) using aria-labelledby. Labels are not required for image, submit, reset, button, or hidden form controls.
###The Algorithm... in English
An <input> (except types of image, submit, reset, button, or hidden), <select>, or <textarea> does not have a properly associated label text. A properly associated label is:
a <label> element with a for attribute value that is equal to the id of a unique form control
a <label> element that surrounds the form control, does not surround any other form controls, and does not reference another element with its for attribute
a non-empty title attribute, or
a non-empty aria-labelledby attribute.
Standards and Guidelines
1.1.1 Non-text Content (Level A)
1.3.1 Info and Relationships (Level A)
2.4.6 Headings and Labels (Level AA)
3.3.2 Labels or Instructions (Level A)
Icon index

## Steps to Reproduce (for bugs)

If you want to reproduce the error please install WAVE evaluation plugin on your browser and use it on a page  which has mui-datatables component. 

Easiest way is to just open one of your example on code sand box and test using plugin.

## Your Environment

| Tech         | Version |
|--------------|---------|
| Material-UI  | 4.10.1       |
| MUI-datatables  |3.0.1         |
| React        |16.13.1         |
| browser      | Chrome         |

**Comment by riserrad** [comment]
+1 I was working on fixing accessibility bugs on a website my team maintains, and the component has some accessibility bugs - reported by [Accessibility Insights](https://microsoftedge.microsoft.com/addons/detail/accessibility-insights-fo/ghbhpcookfemncgoinjblecnilppimih) - that I don't see options to fix it in the component.

I'm also new to React which makes it even more difficult. The errors that it finds:

![image](https://user-images.githubusercontent.com/874409/219827857-f79da3ac-b805-48c4-b65f-85e49e9659fb.png)

**Comment by msftedad** [comment]
Hi @riserrad, Any update on this issue?
